### ISSUE TITLE
Why does the physnet model need DiffNormalized?

### ISSUE THREAD

**Issue by iRuizeWu** [ubicomplab/rPPG-Toolbox#136]
Yu Z, Li X, Zhao G. Remote Photoplethysmograph Signal Measurement from Facial Videos Using Spatio-Temporal Networks[M]. arXiv, 2019.

I can't find anything about DiffNormalized in this article.

**Comment by yahskapar** [comment]
I don't think the full preprocessing steps were revealed in the paper you pointed out to that presented PhysNet. Furthermore, as the authors did not directly release their end-to-end code to reproduce the results in their paper, it's very difficult to know exactly what might have been done in terms of preprocessing steps. All that is mentioned in the "Training Settings" portion of the paper is that, after face cropping, the frames are normalized to 128x128, which as far as I know was tried in the past with the code in this repo and did not yield reasonable results despite the model implementation itself being correct.

@xliucs may have more helpful context to provide, but I think the usage of normalized difference frames is useful for this repo in the sense of allowing a more fair comparison with other neural methods (e.g., TS-CAN, DeepPhys) which utilize normalized difference frames.

**Comment by iRuizeWu** [comment]
Thanks so much. And another question: Estimating heart rate based on rPPG signals consumes a lot of CPU resources.

**Comment by girishvn** [comment]
Hi @iRuizeWu, 

As @yahskapar outlined, the use of normalized difference frames are useful in providing fair comparison w/ other PPG models such as TSCAN/DeepPhys. Additionally, these difference frames allow for some inter-frame information sharing often resulting in a more robust temporal embedding, and have been shown to better help the model learn CHANGES in skin color (caused by the pulse signal).

If you are interested in exploring the model w/ raw frames (as opposed to difference frames) this change can be made in the config files by setting both the data and label types to `Standardized`.

In regards to CPU-burdensome heart-rate estimation: 
Can you specify how the HR estimation time compares to your preprocessing / training / testing time? Depending on the video lengths the estimations may often take several minutes.

**Comment by iRuizeWu** [comment]
Thank you very much for your explanation of DiffNormalized.

I ran the test on the PURE dataset and reasoned with a 3090 sheet, which took 186.89 seconds; However, it took more than 30 minutes to estimate the heart rate based on the rPPG signal.

All experiments are conducted according to the readme.

Have you had this problem?

**Comment by iRuizeWu** [comment]
I found np.linalg.inv in the `_detrend` function in `rPPG-Toolbox/evaluation/post_process.py` to be very time consuming, where the shape is (1920,1920) according to the printed log.

**Comment by xliucs** [comment]
Hi Yes, as PhysNet did not mention details about their preprocessing in the paper, we were not able to reproduce reasonable results on PURE/UBFC by using raw frames. We were only able to get these reasonable results with DiffNormalizaed frames as inputs which I think it is kinda standard in rPPG tasks. Please let us know if you have any concersn. Thanks.

**Comment by yahskapar** [comment]
@iRuizeWu can you create a separate issue regarding your performance observations with more hardware details (e.g., what sort of computing environment you have in terms of CPU resources, GPU resources, RAM, and so on)?
